### PR TITLE
Expanding description for SparsePolynomial::from_coefficients_vec

### PR DESCRIPTION
**Pull request by msmirnov18**
## Description

closes: #XXXX

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (master)
- [ ] Linked to GitHub issue with discussion and accepted design OR have an explanation in the PR that describes this work.
- [ ] Wrote unit tests
- [ ] Updated relevant documentation in the code
- [ ] Added a relevant changelog entry to the `Pending` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the GitHub PR explorer

### CODE REVIEW

File: poly/src/polynomial/univariate/sparse.rs [hunk 1]
```diff
@@ -233,6 +233,26 @@ impl<F: Field> SparsePolynomial<F> {
...
+    /// pub struct FieldConfig;
+    /// pub type FF = Fp64<MontBackend<FieldConfig, 1>>;
+    ///
+    /// // Creating the polynomial 2*x
+    /// let p1 = SparsePolynomial::from_coefficients_vec(vec![(1, FF::from(2))]);
+    /// // Creating the polynomial x + x
+    /// let p2 = SparsePolynomial::from_coefficients_vec(vec![(1, FF::from(1)), (1, FF::from(1))]);
+    /// // Checking if x + x is equal to 2*x leads to panick.
+    /// assert_eq!(p1, p2);
+    /// ```
```
Comment (mmagician) [inline poly/src/polynomial/univariate/sparse.rs:255]: It feels a bit weird to define a negative example like this one. The expanded description should suffice

File: poly/src/polynomial/univariate/sparse.rs [hunk 2]
```diff
@@ -233,6 +233,26 @@ impl<F: Field> SparsePolynomial<F> {
     }
 
     /// Constructs a new polynomial from a list of coefficients.
+    /// Since the function does not automatically combine like terms, the input *has to be* in the
+    /// correct form: no multiple monomials of the same degree.
```
Comment (mmagician) [inline poly/src/polynomial/univariate/sparse.rs:237]:
```suggestion
    /// The function does not combine like terms and so multiple monomials
    /// of the same degree are ignored.
```

File: poly/src/polynomial/univariate/sparse.rs (hunk 1 above)
Comment (mmagician) [reply poly/src/polynomial/univariate/sparse.rs:255]: (side note for the future: the doctests actually have to pass and no panics are allowed. The alternative would be `assert_ne`)

File: poly/src/polynomial/univariate/sparse.rs (hunk 2 above)
Comment (msmirnov18) [reply poly/src/polynomial/univariate/sparse.rs]: I have updated the description according to your suggestions.

Comment (msmirnov18) [reply poly/src/polynomial/univariate/sparse.rs]: I agree that having a negative example in the docs does feel weird.

**Review by mmagician — APPROVED** [review]

**Pull request merged**